ComponentKitTests/Scope/CKDetectComponentScopeCollisionsTests.mm

```diff
@@ -0,0 +1,443 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <XCTest/XCTest.h>
+
+#import <OCMock/OCMock.h>
+
+#import <ComponentKit/CKComponentSubclass.h>
+#import <ComponentKit/CKCompositeComponent.h>
+#import <ComponentKit/CKComponentScope.h>
+#import <ComponentKit/CKComponentInternal.h>
+#import <ComponentKit/CKComponentScopeFrame.h>
+#import <ComponentKit/CKComponentScopeRoot.h>
+#import <ComponentKit/CKDetectComponentScopeCollisions.h>
+
+#pragma mark - Test Components
+
+@interface CKOneComponent : CKComponent
+@end
+
+@implementation CKOneComponent
+@end
+
+@interface CKTwoComponent : CKComponent
+@end
+
+@implementation CKTwoComponent
+@end
+
+@interface CKThreeComponent : CKComponent
+@end
+
+@implementation CKThreeComponent
+@end
+
+@interface CKFourComponent : CKComponent
+@end
+
+@implementation CKFourComponent
+@end
+
+@interface CKFiveComponent : CKComponent
+@end
+
+@implementation CKFiveComponent
+@end
+
+@interface CKSixComponent : CKComponent
+@end
+
+@implementation CKSixComponent
+@end
+
+@interface CKSevenComponent : CKComponent
+@end
+
+@implementation CKSevenComponent
+@end
+
+@interface CKEightComponent : CKComponent
+@end
+
+@implementation CKEightComponent
+@end
+
+@interface CKNineComponent : CKComponent
+@end
+
+@implementation CKNineComponent
+@end
+
+@interface CKTenComponent : CKComponent
+@end
+
+@implementation CKTenComponent
+@end
+
+@interface CKElevenComponent : CKComponent
+@end
+
+@implementation CKElevenComponent
+@end
+
+@interface CKTwelveComponent : CKComponent
+@end
+
+@implementation CKTwelveComponent
+@end
+
+@interface CKThirteenComponent : CKComponent
+@end
+
+@implementation CKThirteenComponent
+@end
+
+#pragma mark - Tests
+
+@interface CKDetectComponentScopeCollisionsTests : XCTestCase
+
+@end
+
+/* This is the tree we're going to generate for the tests.
+ *
+ *   _____1______
+ *  /     |      \
+ *  2 ____3____  4
+ *   /    |    \
+ *   5  __6   __7__
+ *  /   |    /     \
+ *  8   9    10    11
+ *          /  \
+ *         12  13
+ */
+
+@implementation CKDetectComponentScopeCollisionsTests
+
+- (void)testNoCollisionsFound
+{
+  id componentMock1 = OCMClassMock([CKOneComponent class]);
```

Comment:
Sorry @eromanc but we may need to fix the tests. `OCMClassMock` looks like it may not be available with older versions of OCMock. Mind changing this to using `+[OCMockObject mockForClass:]`?